docs/user/moe.md

```diff
@@ -0,0 +1,186 @@
+# Mixture-of-Experts
+
+This tutorial covers the basics of Decentralized Mixture of Experts (DMoE).
+From the infrastructure standpoint, DMoE consists of two parts: experts hosted on peer devices, and a gating/routing function that assigns input to one of these experts.
+
+## Host experts with a server
+
+`hivemind.moe.Server` hosts one or several experts (PyTorch modules) for remote access. These experts are responsible for
+most of the model parameters and computation. The server can be started using either Python or
+[a shell script](https://github.com/learning-at-home/hivemind/blob/master/hivemind/hivemind_cli/run_server.py). We'll use the shell
+for now. To host a server with default experts, run this in your shell:
+
+```sh
+hivemind-server --expert_cls ffn --hidden_dim 512 --num_experts 5 --expert_pattern "expert.[0:5]" \
+                --listen_on 0.0.0.0:1337 --dht_port 1338
+# note: if you omit listen_on and/or dht_port, they will be chosen automatically and printed to stdout.
+```
+
+<details style="margin-top:-24px; margin-bottom: 16px;">
+  <summary><i>Console outputs</i></summary>
+
+```sh
+[2021/07/15 18:52:01.424][INFO][moe.server.create:156] Running DHT node on ['/ip4/127.0.0.1/tcp/42513/p2p/QmacLgRkAHSqdWYdQ8TePioMxQCNV2JeD3AUDmbVd69gNL'], initial peers = []
```

Comment:
The port differs from 1338, are you sure this is the correct output?

Reply:
fixd